pkg/service/service.go

```diff
@@ -827,6 +824,9 @@ func (s *Service) UpdateBackendsState(backends []*lb.Backend) error {
 				if b.L3n4Addr.String() != updatedB.L3n4Addr.String() {
 					continue
 				}
+				if b.State == updatedB.State {
+					break
```

Comment:
This should be `continue`?

Reply:
@aditighag I do not think so. Note this change is not strictly relatd to the one just above (the above was supposed to skip based on the `s.backendByHash[hash]` state, but this one compares `b := range info.backends`

This one makes sense (I believe), because it's after `b.L3n4Addr.String() != updatedB.L3n4Addr.String()`. The execution should go past this condition only for the ONE (correct me if I'm wrong) backend that can match from the backends list and therefore if it's state is already correct, the `UpdateBackendsState` (as suggested by the name) should have nothing to do.

BUT I was actually considering if it shouldn't be:
`if b.State == updatedB.State && b.Preferred == updatedB.Preferred {` .... though I'm not sure the API actually permits the change of just Preffered (not sure how you would even do it).

Reply:
>The execution should go past this condition only for the ONE (correct me if I'm wrong) backend that can match from the backends list

But `UpdateBackendsState(backends []*lb.Backend)` takes a list of backends as the arg, so you should check for state updates for all the backends in the list.

Reply:
Ah! That's right. I mixed up the loops.